db/compaction_picker.cc

```diff
@@ -1373,6 +1356,39 @@ bool LevelCompactionBuilder::PickFileToCompact() {
   return start_level_inputs_.size() > 0;
 }
 
+bool LevelCompactionBuilder::PickL0ToBaseCompaction() {
```

Comment:
One thing we sometimes do currently and don't do with the new change is that, if the oldest file can do a trigger move to L1 (assuming we are lucky enough that the oldest one happens to be the largest), and some newer file will disqualify that, the trivial move will still trigger and move the oldest file. It may be a good idea to get it back.

This was done as a side effect of this approach: pick the oldest file, and file overlapping files in L1. Expand all L0 files that overlap with those L1 files without expanding further L1 files. In this case, if the oldest L0 file is inserted into a hotspot position, then we only compact this L0 file with one L1 file and nothing more. By aggressively adding all L0 files, this optimization is removed.